cf_xarray/accessor.py

```diff
@@ -88,9 +89,11 @@ def apply_mapper(
     results for a good key.
     """
 
-    if not isinstance(key, str):
+    if not isinstance(key, collections.abc.Hashable):
```

Comment:
```suggestion
    if not isinstance(key, Hashable):
```

would this just work?